pkg/certificates/cert_controller.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package certificates
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"reflect"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/open-cluster-management/addon-framework/pkg/addonmanager"
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/config"
+	"github.com/open-cluster-management/multicluster-observability-operator/pkg/util"
+)
+
+const (
+	restartLabel = "cert/time-restarted"
+)
+
+func Start() {
+
+	// setup ocm addon manager
+	addonMgr, err := addonmanager.New(ctrl.GetConfigOrDie())
+	if err != nil {
+		log.Error(err, "Failed to init addon manager")
+		os.Exit(1)
+	}
+	agent := &ObservabilityAgent{}
+	addonMgr.AddAgent(agent)
+	addonMgr.Start(context.TODO())
+
+	kubeClient, err := kubernetes.NewForConfig(ctrl.GetConfigOrDie())
+	if err != nil {
+		log.Error(err, "Failed to create kube client")
+		os.Exit(1)
+	}
+	watchlist := cache.NewListWatchFromClient(kubeClient.CoreV1().RESTClient(), "secrets", config.GetDefaultNamespace(),
+		fields.Everything())
+	_, controller := cache.NewInformer(
+		watchlist,
+		&v1.Secret{},
+		time.Minute*60,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				restartPods(*kubeClient, *obj.(*v1.Secret))
```

Comment:
the rbac-query-proxy and  observatorium-api  will be restarted for the first time when the secrets are created. is it  desired behaviour?

Reply:
in function updateDeployLabel, the restart label will only be injected when secret is created after deployment. So for first time,  there will be no restart.

Reply:
👍 You can follow the same way of cert-manager does.